src/test/datascience/notebook/helper.ts

```diff
@@ -792,7 +792,9 @@ export async function waitForEvent<T>(
     timeout = env.uiKind === UIKind.Desktop ? 5000 : 15000
 ): Promise<T> {
     setTimeout(() => {
-        throw new Error(`Debugging Timeout: ${eventType} message took too long to arrive`);
+        clearTimeout(timeout);
+        console.log(`Test failing --- Debugging Timeout: ${eventType} message took too long to arrive`);
+        return Promise.reject();
```

Comment:
This is being returned from `setTimeout` and not going anywhere, you need a promise returned from `waitForEvent`.

Reply:
I'm not sure what happend to the suggestion I made before? waitForCondition could be used here. It would make this function super simple.